mu-plugins/rest-api/endpoints/class-wporg-plugins-locale-banner-controller.php

```diff
@@ -0,0 +1,191 @@
+		$current_locale = get_locale();
+		$current_gp_locale = \GP_Locales::by_field( 'wp_locale', $current_locale );
+
+		// Build a list of WordPress locales which we'll suggest to the user.
+		$suggest_locales = array_values( array_intersect( get_locale_from_header(), get_all_valid_locales() ) );
+
+		$suggestion_links = [];
+		foreach ( $suggest_locales as $locale ) {
+			$language = \GP_Locales::by_field( 'wp_locale', $locale )->native_name;
+			$suggestion_links[ $locale ] = sprintf(
+				'<a href="https://%s.wordpress.org%s">%s</a>',
+				$locale_subdomain_assoc[ $locale ]->subdomain,
+				esc_url( get_site()->path ),
+				$language
+			);
+		}
+
+		$suggest_string = '';
+
+		unset( $suggestion_links[ $current_locale ] );
+
+		if ( ! empty( $suggestion_links ) ) {
+			$output_locale = key( $suggestion_links );
+			switch_to_locale( $output_locale );
+			$suggest_string = sprintf(
+				// translators: %s: List of links to plugin directory in other locales.
+				__( 'The plugin directory is also available in %s.', 'wporg' ),
+				wp_sprintf_l( '%l', $suggestion_links )
+			);
+		}
+
+		// Return more information if this is a debug request.
+		if ( ! empty( $request['debug'] ) ) {
+			return new \WP_REST_Response(
+				array(
+					'currentLocale' => $current_locale,
+					'suggestions' => $suggest_locales,
+					'message' => $suggest_string,
+				)
+			);
+		}
+
+		// The result should be a raw text response.
+		add_filter( 'rest_pre_echo_response', array( $this, 'send_plain_text' ) );
+		return new \WP_REST_Response( $suggest_string );
+	}
+
+	/**
+	 * Get banner for single plugins.
+	 */
+	public function get_response_for_item( $request ) {
+		// This has already been validated by `validate_callback`.
+		$plugin_slug = $request['slug'];
+
+		if ( ! defined( 'GLOTPRESS_LOCALES_PATH' ) ) {
+			return;
+		}
+
+		require_once GLOTPRESS_LOCALES_PATH;
+
+		$locale_subdomain_assoc = get_all_locales_with_subdomain();
+		$current_locale = get_locale();
+		$current_gp_locale = \GP_Locales::by_field( 'wp_locale', $current_locale );
+		$translated_locales = get_translated_locales( 'plugin', $plugin_slug );
+
+		// Build a list of WordPress locales which we'll suggest to the user.
+		$suggest_locales = array_values( array_intersect( get_locale_from_header(), $translated_locales ) );
+
+		$suggestion_links = [];
+		foreach ( $suggest_locales as $locale ) {
+			$language = \GP_Locales::by_field( 'wp_locale', $locale )->native_name;
+			$suggestion_links[ $locale ] = sprintf(
+				'<a href="https://%s.wordpress.org%s">%s</a>',
+				$locale_subdomain_assoc[ $locale ]->subdomain,
+				esc_url( get_site()->path . $plugin_slug . '/' ),
+				$language
+			);
+		}
+
+		$suggest_string = '';
+
+		unset( $suggestion_links[ $current_locale ] );
+
+		// If we're on a rosetta site, and the plugin is not translated, the message should ask for help.
+		if ( 'en_US' !== $current_locale && $current_gp_locale && ! in_array( $current_locale, $translated_locales ) ) {
```

Comment:
I'm curious why `$current_gp_locale` needs to be checked in this if statement. That's not to question its need, but as a future developer looking at this code, it isn't clear to me why this would be falsey and what should happen in that case. Minor.

Reply:
To be honest I copied this from the current code, and I'm not totally sure when it would be false— I would _assume_ all rosetta sites correspond to a GP locale.

Well, it looks like it's false on `https://test.wordpress.org/plugins/wp-json/wporg-plugins/v1/locale-banner/recipe-block/`, so there's at least one place (test's locale uses `de`, but it should be `de_DE`). I don't know if any non-test sites trigger it.